src/extensions/taction/tactionext.cpp

```diff
@@ -0,0 +1,644 @@
+
+            if (hptr->Class->House != taction.Data.House) {
+                hptr->IsDefeated = true;
+            }
+        }
+    }
+
+    return true;
+}
+
+
+/**
+ *  #issue-965
+ *
+ *  Makes the "Loser is" trigger action set the IsDefeated flag on the
+ *  losing house in multiplayer.
+ *
+ *  @author: Rampastring
+ */
+bool TActionClassExtension::Do_LOSE(TActionClass& taction, HouseClass* house, ObjectClass* object, TriggerClass* trig, const Cell& cell)
+{
+    /**
+     *  Flag the player as won or lost, like in the original code.
+     */
+    if (taction.Data.House != PlayerPtr->Class->House) {
+        PlayerPtr->Flag_To_Win();
+    } else {
+        PlayerPtr->Flag_To_Lose();
+    }
+    
+    if (Session.Type != GAME_NORMAL) {
+
+        /**
+         *  Mark all losers as defeated.
+         */
+        for (int i = 0; i < Houses.Count(); i++) {
+            HouseClass* hptr = Houses[i];
+
+            if (hptr->Class->House == taction.Data.House) {
+                hptr->IsDefeated = true;
+            }
+        }
+    }
+
+    return true;
+}
+
+
+/**
+ *  #issue-33
+ *
+ *  Fixes the issue where TACTION_DESTROY_TRIGGER would cause the game to
+ *  crash if it's used to destroy triggers of its own type.
+ *
+ *  @author: ZivDero
+ */
+bool TActionClassExtension::Do_DESTROY_TRIGGER(TActionClass& taction, HouseClass* house, ObjectClass* object, TriggerClass* trig, const Cell& cell)
+{
+    if (taction.Trigger != nullptr) {
+        int count = Triggers.Count();
+
+        for (int index = count - 1; index >= 0; index--) {
+            if (Triggers[index]->Class == taction.Trigger) {
+                Triggers[index]->Mark_To_Die();
+            }
+        }
+    }
+    return true;
+}
+
+
+/**
+ *  #issue-299
+ *
+ *  Fixes the issue with the current difficulty not being checked
+ *  when enabling triggers.
+ *
+ *  @author: CCHyper
+ */
+bool TActionClassExtension::Do_ENABLE_TRIGGER(TActionClass& taction, HouseClass* house, ObjectClass* object, TriggerClass* trig, const Cell& cell)
+{
+    /**
+     *  This is direct port of the code from Red Alert 2, which looks to fix this issue.
+     */
+    if (taction.Trigger != nullptr) {
+        for (int index = 0; index < Triggers.Count(); index++) {
+            if (Triggers[index]->Class == taction.Trigger) {
+                bool really_enable = true;
+
+                /**
+                 *  Set this trigger to be disabled if it is marked as disabled
+                 *  for this current mission difficulty.
+                 */
+                if (Scen->Difficulty == DIFF_EASY && !Triggers[index]->Class->IsEnabledEasy) {
+                    really_enable = false;
+                } else if (Scen->Difficulty == DIFF_NORMAL && !Triggers[index]->Class->IsEnabledMedium) {
+                    really_enable = false;
+                } else if (Scen->Difficulty == DIFF_HARD && !Triggers[index]->Class->IsEnabledHard) {
+                    really_enable = false;
+                }
```

Comment:
Need to check CDifficulty instead of Difficulty, and always enable in multiplayer.